### PR TITLE
[FEATURE] Support multiple conditions for a single file

### PR DESCRIPTION
**Pull request by eliashaeussler**
This PR introduces support for multiple file conditions for a single file as configured in the `processSourcesFiles` and `processSharedSourceFiles` steps. This is especially useful in combination with the `target` file configuration, e.g. if a single file must be located differently, depending on a configured template variable.

Example:

TYPO3 CMS v12 changed file locations for configuration files:

* `public/typo3conf/LocalConfiguration.php` → `config/system/settings.php`
* `public/typo3conf/AdditionalConfiguration.php` → `config/system/additional.php`

Since file contents are the same across different framework versions, only the file location is relevant when processing source files. This PR allows to override the target file location in case v12 has been selected as target version for TYPO3 CMS. The source file itself can be left unchanged.

### CODE REVIEW

File: docs/processing-build-steps.md [hunk 1]
```diff
@@ -113,11 +113,28 @@ If a file has the `.twig` extension, it is first processed by the Twig renderer.
...
+on a given file condition.
+
+#### Multiple conditions
+
+It is possible to define more than one file condition for a single file. This is
+useful in combination with the configuration of a file target, e.g. if the file
+is expected on different locations for different versions of a configured framework.
+If more than one file condition evaluates to `true`, the first configured target is
+always used.
+
```
Comment (mteu) [inline docs/processing-build-steps.md:137]: Thanks, @eliashaeussler! Although the documentation itself is perfectly comprehensible I fear that we're starting to miss out on useful opportunities to showcase the powerful capabilities of the project-builder.

In this specific PR I reckon we offer a few more examples to have users grasps the underlying potential. I understand that this ask isn't an easy one because we should identify configuration examples that work for most developers despite their actual project. 

Now, I understand that the project-builder is designed to offer a very generic approach of kickstarting composer-based  projects. However, the actual implementation requires a template, of course. At this very moment I only see TYPO3 and Symfony implementations making use of the project builder. This might hopefully change in the future.

**TL;DR;**
Please add an example. Similiar to the PR description, perhaps.

Comment (mteu) [reply docs/processing-build-steps.md:137]: Since the documentation is bigger issue we'll take that into account here #50.
